src/librustc/traits/query/type_op/custom.rs

```diff
@@ -55,3 +66,35 @@ where
         write!(f, "{}", (self.description)())
     }
 }
+
+/// Executes `op` and then scrapes out all the "old style" region
+/// constraints that result, creating query-region-constraints.
+fn scrape_region_constraints<'gcx, 'tcx, R>(
```

Comment:
Isn't this quite similar to what's happening in `librustc_traits/type_op_*.rs`? Could they use the same abstraction?